<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Pipeline to construct evolutionary trees • outsider</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Pipeline to construct evolutionary trees">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">outsider</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/outsider.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/available.html">Available modules</a>
    </li>
    <li>
      <a href="../articles/phylogenetic_pipeline.html">Pipeline to construct evolutionary trees</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Pipeline to construct evolutionary trees</h1>
            
            <h4 class="date">2019-10-17</h4>
      
      
      <div class="hidden name"><code>phylogenetic_pipeline.Rmd</code></div>

    </div>

    
    
<p>The whole point of <code>outsider</code> is to bring lots of programs together into a single place to allow the development of analysis pipelines. Let’s demonstrate this process by creating a simple pipeline for constructing an evolutionary tree. We will generate a tree in three steps: obtain orthologous DNA sequences, align these sequences and then estimate a tree.</p>
<p>In our example, we will keep it simple and fast and generate a tree of <a href="https://en.wikipedia.org/wiki/Cytochrome_b">CytB</a> sequences in the primate genus of Night Monkeys, <strong>Aotus</strong>.</p>
<div id="phylotar" class="section level1">
<h1 class="hasAnchor">
<a href="#phylotar" class="anchor"></a>phylotaR</h1>
<p>To get us started we need DNA sequences that represent the same gene region, i.e. sequences that are orthologous. We can obtain our sequences quickly using the example data from the R package <a href="https://github.com/ropensci/phylotaR"><code>phylotaR</code></a>.</p>
<p><code>phylotaR</code> provides a pipeline for identifying orthologous sequences from a reputable online biological sequence database. The package comes with a few pre-calculated example sequences. We can extract sequence clusters that represent the gene region <a href="https://en.wikipedia.org/wiki/Cytochrome_b">cytb</a> with the following script.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">aotus &lt;-<span class="st"> </span><span class="ot">NULL</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(phylotaR)
<span class="co"># Example data</span>
<span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"aotus"</span>)
<span class="co"># Generate summary of identified clusters</span>
smmry &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/summary.html">summary</a></span>(aotus)
<span class="co"># Extract cluster with 'cytb' in feature name</span>
cytb &lt;-<span class="st"> </span>smmry<span class="op">$</span>ID[<span class="kw"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span>(<span class="st">'cytb'</span>, smmry<span class="op">$</span>Feature))[<span class="dv">1</span>]]
cytb &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/phylotaR/man/drop_clstrs.html">drop_clstrs</a></span>(aotus, cytb)
<span class="co"># Reduce cluster to just one sequence per taxon</span>
cytb &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/phylotaR/man/drop_by_rank.html">drop_by_rank</a></span>(cytb, <span class="dt">n =</span> <span class="dv">1</span>)
<span class="co"># Get taxonomic IDs for taxa in cluster</span>
txids &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/phylotaR/man/get_txids.html">get_txids</a></span>(cytb, <span class="dt">cid =</span> cytb<span class="op">@</span>cids[[<span class="dv">1</span>]])
<span class="co"># Convert IDs to species names</span>
sp_nms &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/phylotaR/man/get_tx_slot.html">get_tx_slot</a></span>(cytb, txids, <span class="dt">slt_nm =</span> <span class="st">'scnm'</span>)
sp_nms &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/grep.html">sub</a></span>(<span class="st">' '</span>, <span class="st">'_'</span>, sp_nms)
<span class="co"># Write out</span>
<span class="kw"><a href="https://rdrr.io/pkg/phylotaR/man/write_sqs.html">write_sqs</a></span>(cytb, <span class="st">'aotus_cytb.fasta'</span>, <span class="dt">sq_nm =</span> sp_nms, <span class="dt">sid =</span> <span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(sp_nms))
<span class="co"># What do the first 50 lines of the file look like?</span>
<span class="kw"><a href="https://rdrr.io/r/base/cat.html">cat</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/readLines.html">readLines</a></span>(<span class="st">'aotus_cytb.fasta'</span>, <span class="dt">n =</span> <span class="dv">50</span>), <span class="dt">collapse =</span> <span class="st">'</span><span class="ch">\n</span><span class="st">'</span>))</code></pre></div>
<pre><code>## &gt;Aotus_nancymaae
## atgacttctccccgcaaaacacacccactaacaaagatcattaacgaatcattcattgatctacccacaccacccaacat
## ttcctcctgatgaaattttggctcactcttaggcatttgcctaattattcaaatcaccaccggcctgttcttagctatac
## actacacaccagatacctcaaccgccttctcctccgtcgcccatatcacccgagacgtcaactatggctgaataattcgc
## tacatacatgccaacggtgcttccatattcttcgtatgcctttttctccacattggtcgaggactttactatggatcctt
## tctttctctgaagacttgaaatatcggtaccatcctactacttacaaccatagccacagcattcataggctatgttcttc
## catgaggccaaatatcattctgaggggctacagtaattacaaatcttttatcagccattccctatatcggatctgacctt
## gtacaatgaatttgaggtggcttctcagtagataaagccactctcacacgattctttacttttcactttatcttaccctt
## cattatcgcagccctagcaactatccatctattatttctgcatgaaacaggatcaagtaacccatcaggaataacatctg
## accccgacaaaatcacatttcacccctattatacagctaaagacattctaggattaatctttcttctcttatccctaata
## agcctaaccctatttatacccgaccttttaaccgacccagataattatacactggctaatcccctcaacactccacccca
## catcaagccagagtgatattttctatttgcatacgcaatcctacgatctatccctaataaacttggaggagttctagccc
## tagtactttctattttaattctaatagttatccctatactacatctctccaaacaacaaagcataatatttcgacccatc
## actcaaattctattctgaactctagtagctgacctactaactctcacatgaattggaggccaaccggttgaatacccctt
## cgtaaccattggccaaaccgcatccattacatacttcttcattattattatcctaatgcccctttccgcctcaatcgaaa
## atatattacttaaatgataa
## 
## &gt;Aotus_azarai
## atgacctccccccgcaaaacacacccactagcaaagattattaacgaatcattcatcgatctccccacaccatccaacat
## ttcctcttgatgaaattttggctcactcttaggcatttgcctaatcattcaaatcaccaccggcctgttcttagctatac
## attacacaccagatacctcaactgccttctcctccgtcgctcatatcacccgagacgttaactatggctgaataattcgc
## tatatacatgccaacggcgcttccatattcttcgtatgcctttttctccatattggccgaggactttactatggatcttt
## cctttttctgaagacttgaaatatcggtattatcctactacttacaaccatagccacagcattcataggctatgttcttc
## catgaggccaaatatcattctgaggggccacagtaattacaaaccttctatcagctatcccctatatcgggtctgacctt
## gtacaatgaatttgaggtggcttctcagtagataaagccactctcacacgattctttacttttcactttatcttaccctt
## tattatcgcagccctagcaactattcacctcttatttctacatgaaacaggatcaagcaacccatcaggaataacatctg
## accccgacaaagtcacattccacccctattatacagctaaggatattctaggattaatctttcttctcttatccctaata
## agcctaaccctatttatacccgaccttctaacggacccagataattatacactagccaaccctctcaacaccccgcctca
## cattaagccagagtggtattttctatttgcatacgcaattctacgatctatccctaataaacttggaggagtactagccc
## tagtactttctatcctaatcttaatggctatccccgtactacatttctccaaacagcaaagtataatatttcgacccatt
## actcaaattctattctgagctctggtagctgacctactaactctcacgtgaattggaggtcaaccagttgagtacccctt
## cgtaaccattggccagaccgcatccattatatacttcttcattattattaccctaataccccttttcgccttaattgaaa
## ataaattacttaaatgatag
## 
## &gt;Aotus_griseimembra
## atgacttctccccgcaaaacacacccactaacaaagatcattaacgaatcattcattgatctacccacaccacccaacat
## ttcctcctgatgaaattttggctcactcttaggcatttgcctaattattcaaatcaccaccggcctgttcttagctatac
## actacacaccagatacctcaaccgccttctcctccgtcgcccatatcacccgagacgtcaactatggctgaataattcgc
## tacatacatgccaacggtgcttccatattcttcgtatgcctttttctccacattggtcgaggactttactatggatcctt
## cctttctctgaagacttgaaatatcggtaccatcctactacttacaaccatagccacagcattcataggctatgttcttc
## catgaggccaaatatcattctgaggggctacagtaattacaaatcttatatcagccattccctatatcggatctgacctt
## gtacaatgaatttgaggtggcttctcagtagataaagccactctcacacgattctttacttttcactttatcttaccctt
## cattatcgcagccctagcaactatccatctattatttctgcatgaaacaggatcaagtaacccatcaggaataacatctg
## accccgacaaaatcacatttcacccctattatacagctaaagacattctagggttaatctttcttctcttatccctaata
## agcctaaccctatttatacccgaccttttaaccgacccagataattatacactggctaatcccctcaacactccacccca
## catcaagccagagtgatattttctatttgcatacgcaattctacgatctatccctaataaacttggaggagttctagccc
## tagtactttctattttaattctaatagttatccctatactacatctctccaaacaacaaagcataatatttcgacccatc
## actcaaattctattctgaactctagtagctgacctactaactctcacatgaattggaggccaaccggttgaatacccctt
## cgtaaccattggccaaaccgcatccattacatacttcttcattattattatcctaatgcccctttccgcctcaatcgaaa
## atatattacttaaatgataa</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/base/cat.html">cat</a></span>(<span class="st">'...'</span>)</code></pre></div>
<pre><code>## ...</code></pre>
</div>
<div id="alignment" class="section level1">
<h1 class="hasAnchor">
<a href="#alignment" class="anchor"></a>Alignment</h1>
<p>Now we have written our sequences to a text-based <code>.fasta</code> file, we need to align them! To do this, we can use <a href="https://mafft.cbrc.jp/alignment/software/"><code>mafft</code></a>. We first need to install the module and then import the key function, <code>mafft</code>. After doing this, we can call the <code>mafft</code> program using the same arguments that we would if it were calling the program via command-line. In our case we can simple run our alignment using the ‘–auto’ parameter.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(outsider)</code></pre></div>
<pre><code>## ----------------
## outsider v 0.1.0
## ----------------
## - Security notice: be sure of which modules you install</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Install module</span>
repo &lt;-<span class="st"> 'dombennett/om..mafft'</span>
<span class="kw"><a href="../reference/module_install.html">module_install</a></span>(<span class="dt">repo =</span> repo, <span class="dt">force =</span> <span class="ot">TRUE</span>)
<span class="co"># Import the function</span>
mafft &lt;-<span class="st"> </span><span class="kw"><a href="../reference/module_import.html">module_import</a></span>(<span class="dt">fname =</span> <span class="st">'mafft'</span>, <span class="dt">repo =</span> repo)
<span class="co"># Run our program with normal mafft arguments</span>
<span class="co"># Note: all arguments must be separate character variables.</span>
<span class="kw">mafft</span>(<span class="dt">arglist =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'--auto'</span>, <span class="st">'aotus_cytb.fasta'</span>, <span class="st">'&gt;'</span>, <span class="st">'alignment.fasta'</span>))</code></pre></div>
<pre><code>## outputhat23=16
## treein = 0
## compacttree = 0
## stacksize: 8192 kb
## generating a scoring matrix for nucleotide (dist=200) ... done
## All-to-all alignment.
##     0 / 8
    1 / 8
    2 / 8
    3 / 8
    4 / 8
    5 / 8
    6 / 8
tbfast-pair (nuc) Version 7.407
## alg=L, model=DNA200 (2), 2.00 (6.00), -0.10 (-0.30), noshift, amax=0.0
## 0 thread(s)
## 
## outputhat23=16
## Loading 'hat3.seed' ... 
## done.
## Writing hat3 for iterative refinement
## generating a scoring matrix for nucleotide (dist=200) ... done
## Gap Penalty = -1.53, +0.00, +0.00
## tbutree = 1, compacttree = 0
## Constructing a UPGMA tree ... 
## 
    0 / 8
## done.
## 
## Progressive alignment ... 
## 
STEP     1 /7 
STEP     2 /7 
STEP     3 /7 
STEP     4 /7 
STEP     5 /7 
STEP     6 /7 
STEP     7 /7 
## done.
## tbfast (nuc) Version 7.407
## alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0
## 1 thread(s)
## 
## minimumweight = 0.000010
## autosubalignment = 0.000000
## nthread = 0
## randomseed = 0
## blosum 62 / kimura 200
## poffset = 0
## niter = 16
## sueff_global = 0.100000
## nadd = 16
## Loading 'hat3' ... done.
## generating a scoring matrix for nucleotide (dist=200) ... done
## 
## 
    0 / 8
## Segment   1/  1    1-1141
## STEP 001-001-0  identical.   
STEP 001-001-1  identical.   
STEP 001-002-0  identical.   
STEP 001-002-1  identical.   
STEP 001-003-0  identical.   
STEP 001-003-1  identical.   
STEP 001-004-0  identical.   
STEP 001-004-1  identical.   
STEP 001-005-0  identical.   
STEP 001-005-1  identical.   
STEP 001-006-0  identical.   
STEP 001-006-1  identical.   
STEP 001-007-1  identical.   
STEP 002-007-1  identical.   
STEP 002-006-0  identical.   
STEP 002-006-1  identical.   
## Converged.
## 
## done
## dvtditr (nuc) Version 7.407
## alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0
## 0 thread(s)
## 
## 
## Strategy:
##  L-INS-i (Probably most accurate, very slow)
##  Iterative refinement method (&lt;16) with LOCAL pairwise alignment information
## 
## If unsure which option to use, try 'mafft --auto input &gt; output'.
## For more information, see 'mafft --help', 'mafft --man' and the mafft page.
## 
## The default gap scoring scheme has been changed in version 7.110 (2013 Oct).
## It tends to insert more gaps into gap-rich regions than previous versions.
## To disable this change, add the --leavegappyregion option.</code></pre>
</div>
<div id="phylogeny" class="section level1">
<h1 class="hasAnchor">
<a href="#phylogeny" class="anchor"></a>Phylogeny</h1>
<p>We can repeat the same process as we did for <code>mafft</code> but instead for <a href="https://github.com/stamatak/standard-RAxML"><code>RAxML</code></a> – a command-line program for estimating evolutionary trees from an alignment of DNA sequences using a model of molecular evolution (GTRGAMMA) with maximum likelihood.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(outsider)
<span class="co"># Install</span>
repo &lt;-<span class="st"> 'dombennett/om..raxml'</span>
<span class="kw"><a href="../reference/module_install.html">module_install</a></span>(<span class="dt">repo =</span> repo, <span class="dt">force =</span> <span class="ot">TRUE</span>)
<span class="co"># Import</span>
raxml &lt;-<span class="st"> </span><span class="kw"><a href="../reference/module_import.html">module_import</a></span>(<span class="dt">fname =</span> <span class="st">'raxml'</span>, <span class="dt">repo =</span> repo)
<span class="co"># Run</span>
<span class="kw">raxml</span>(<span class="dt">arglist =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'-m'</span>, <span class="st">'GTRGAMMA'</span>, <span class="st">'-s'</span>, <span class="st">'alignment.fasta'</span>, <span class="st">'-p'</span>, <span class="st">'1234'</span>,
                  <span class="st">'-n'</span>, <span class="st">'aotus_cytb'</span>, <span class="st">'-T'</span>, <span class="st">'2'</span>))</code></pre></div>
<pre><code>## 
## RAxML can't, parse the alignment file as phylip file 
## it will now try to parse it as FASTA file
## 
## 
## 
## Using BFGS method to optimize GTR rate parameters, to disable this specify "--no-bfgs" 
## 
## 
## This is the RAxML Master Pthread
## 
## This is RAxML Worker Pthread Number: 1
## 
## 
## This is RAxML version 8.2.12 released by Alexandros Stamatakis on May 2018.
## 
## With greatly appreciated code contributions by:
## Andre Aberer      (HITS)
## Simon Berger      (HITS)
## Alexey Kozlov     (HITS)
## Kassian Kobert    (HITS)
## David Dao         (KIT and HITS)
## Sarah Lutteropp   (KIT and HITS)
## Nick Pattengale   (Sandia)
## Wayne Pfeiffer    (SDSC)
## Akifumi S. Tanabe (NRIFS)
## Charlie Taylor    (UF)
## 
## 
## Alignment has 74 distinct alignment patterns
## 
## Proportion of gaps and completely undetermined characters in this alignment: 0.00%
## 
## RAxML rapid hill-climbing mode
## 
## Using 1 distinct models/data partitions with joint branch length optimization
## 
## 
## Executing 1 inferences on the original alignment using 1 distinct randomized MP trees
## 
## All free model parameters will be estimated by RAxML
## GAMMA model of rate heterogeneity, ML estimate of alpha-parameter
## 
## GAMMA Model parameters will be estimated up to an accuracy of 0.1000000000 Log Likelihood units
## 
## Partition: 0
## Alignment Patterns: 74
## Name: No Name Provided
## DataType: DNA
## Substitution Matrix: GTR
## 
## 
## 
## 
## RAxML was called as follows:
## 
## raxmlHPC-PTHREADS-SSE3 -m GTRGAMMA -s alignment.fasta -p 1234 -n aotus_cytb -T 2 
## 
## 
## Partition: 0 with name: No Name Provided
## Base frequencies: 0.289 0.285 0.120 0.306 
## 
## Inference[0]: Time 0.131239 GAMMA-based likelihood -2591.962827, best rearrangement setting 5
## 
## 
## Conducting final model optimizations on all 1 trees under GAMMA-based models ....
## 
## Inference[0] final GAMMA-based Likelihood: -2591.962819 tree written to file /working_dir/RAxML_result.aotus_cytb
## 
## 
## Starting final GAMMA-based thorough Optimization on tree 0 likelihood -2591.962819 .... 
## 
## Final GAMMA-based Score of best tree -2591.962819
## 
## Program execution info written to /working_dir/RAxML_info.aotus_cytb
## Best-scoring ML tree written to: /working_dir/RAxML_bestTree.aotus_cytb
## 
## Overall execution time: 0.173492 secs or 0.000048 hours or 0.000002 days</code></pre>
</div>
<div id="visualisation" class="section level1">
<h1 class="hasAnchor">
<a href="#visualisation" class="anchor"></a>Visualisation</h1>
<p>Now, let’s check out our tree! We can use the R package <a href="http://ape-package.ird.fr/">“Analsis of Phylogenetics and Evolution” or <code>ape</code></a> to do this.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(ape)
tree &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span>(<span class="st">'RAxML_bestTree.aotus_cytb'</span>)
<span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(tree)</code></pre></div>
<p><img src="phylogenetic_pipeline_files/figure-html/visualise-1.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#phylotar">phylotaR</a></li>
      <li><a href="#alignment">Alignment</a></li>
      <li><a href="#phylogeny">Phylogeny</a></li>
      <li><a href="#visualisation">Visualisation</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Dom Bennett.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9100.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
